app/main.py

```diff
@@ -1,34 +1,81 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple, end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = self.create_deck()
+        self.count_hits = 0
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def create_deck(self) -> list:
+        decks_for_ship = []
+        start_row, start_column = self.start
+        end_row, end_column = self.end
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+        if self.start == self.end:
+            one_deck = Deck(self.start[0], self.start[1])
+            decks_for_ship.append(one_deck)
+            return decks_for_ship
+
+        if start_row == end_row:
+            for column in range(start_column, end_column + 1):
+                decks_for_ship.append(Deck(start_row, column))
+
+        elif start_column == end_column:
+            for row in range(start_row, end_row + 1):
+                decks_for_ship.append(Deck(row, start_column))
+        return decks_for_ship
+
+    def get_deck(self, row: int, column: int) -> Deck:
+        deck = next((
+            deck for deck in self.decks
+            if deck.row == row and deck.column == column), None
+        )
+        return deck
+
+    def fire(self, row: int, column: int) -> str:
+        message_fire = ""
+        for deck in self.decks:
+            if row == deck.row and column == deck.column:
+                deck.is_alive = False
+                message_fire = "Hit!"
+                self.count_hits += 1
+
+        if self.count_hits == len(self.decks):
+            message_fire = "Sunk!"
+
+        return message_fire
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list) -> None:
```

Comment:
Be more specific in the type annotations and return type annotations where you use `list` to indicate the expected content of `list`.